pkg/apis/greenhouse/v1alpha1/conditions.go

```diff
@@ -140,6 +145,17 @@ func (sc *StatusConditions) SetConditions(conditionsToSet ...Condition) {
 	}
 }
 
+// removeCondition removes the condition with the given type from the list of conditions.
+func removeCondition(conditions []Condition, conditionType ConditionType) []Condition {
+	var newConditions []Condition
+	for _, condition := range conditions {
+		if condition.Type != conditionType {
+			newConditions = append(newConditions, condition)
+		}
+	}
+	return newConditions
```

Comment:
Such common operations on slices are now also covered by the standard library:

```suggestion
	return slices.DeleteFunc(conditions, func(c Condition) bool{
	        return condition.Type == conditionType
	}
```

https://pkg.go.dev/slices#DeleteFunc

Reply:
this is a great suggestion. Fixed in 7bf9dca5644f181ad067ea091821f3a2b36e1174

Reply:
🚀